src/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/ldap/LdapDirectoryIdentifier.cs

```diff
@@ -55,31 +55,7 @@ public LdapDirectoryIdentifier(string[] servers, int portNumber, bool fullyQuali
             PortNumber = portNumber;
         }
 
-        public string[] Servers
-        {
-            get
-            {
-                if (_servers == null)
-                {
-                    return Array.Empty<string>();
-                }
-                
-                var temporaryServers = new string[_servers.Length];
-                for (int i = 0; i < _servers.Length; i++)
-                {
-                    if (_servers[i] != null)
-                    {
-                        temporaryServers[i] = string.Copy(_servers[i]);
-                    }
-                    else
-                    {
-                        temporaryServers[i] = null;
-                    }
-                }
-
-                return temporaryServers;
-            }
-        }
+        public string[] Servers => (_servers == null) ? Array.Empty<string>() : (string[])_servers.Clone();
```

Comment:
I wonder if we should make Clone on an empty array return Empty.

Reply:
Looking through the VM code, might be a bit difficult to normalize to `Array.Empty<T>()`, but it should be fairly straightforward to make it return `this` if it's empty. Seem like a good compromise?

(And heads up - in the case being considered here we're comparing against null anyway, as calling `<null>.Clone()` would null ref.)

Reply:
> Seem like a good compromise?

Yes. Though not for this PR. And I'm now questioning myself and wondering if Clone might be one of those things a dev expects to always return something new.

Reply:
I was inspired by this and opened https://github.com/dotnet/coreclr/issues/23382, which would've allowed the VM to normalize `new T[0]` to a singleton. Jan commented on that issue - it turns out that proposal may have been a step too far. One of his concerns is similar to the one you just voiced re: dev expectations of the `Clone` method.